state/application.go

```diff
@@ -2190,12 +2197,23 @@ func (a *Application) GetScale() int {
 
 // ChangeScale alters the existing scale by the provided change amount, returning the new amount.
 // This is used on CAAS models.
-func (a *Application) ChangeScale(scaleChange int) (int, error) {
+func (a *Application) ChangeScale(scaleChange int, attachStorage []string) (int, error) {
 	newScale := a.doc.DesiredScale + scaleChange
 	logger.Tracef("ChangeScale DesiredScale %v, scaleChange %v, newScale %v", a.doc.DesiredScale, scaleChange, newScale)
 	if newScale < 0 {
 		return a.doc.DesiredScale, errors.NotValidf("cannot remove more units than currently exist")
 	}
+
+	// Convert attach storage ids to storage tags.
+	storageTags := make([]names.StorageTag, len(attachStorage))
+	for i, stor := range attachStorage {
+		if names.IsValidStorage(stor) {
+			storageTags[i] = names.NewStorageTag(stor)
+		} else {
+			return a.doc.DesiredScale, errors.NotValidf("storage name %q", stor)
+		}
+	}
+
```

Comment:
This validation is done earlier in the facade. Like with insertCAASUnitOps, we need to pass in storage tags instead of strings

Reply:
Want to confirm again, because the [previous comment](https://github.com/juju/juju/pull/20184#discussion_r2208907689) said we should pass the string instead of tag into state layer. Here is just a code to convert the id string to tag. Or I misunderstanding your instruction.